source/dagger/gameplay/platformer/platformer_main.h

```diff
@@ -1,12 +1,89 @@
 #pragma once
 
-#include "core/core.h"
+
 #include "core/game.h"
+#include "core/core.h"
```

Comment:
you need to try to use as less includes as possible in .h files.

This thing might be used here:
https://en.wikipedia.org/wiki/Forward_declaration
u are basically adding a prototypes and add includes with implementations in cpp

Reply:
I moved the definition of the methods to the .cpp file , but if I use the forward declaration I can't use the Character struct methods within other .cpp files. So I had to leave some includes in the .h file , although I'm sure there is a better way to do this  , any suggestions?

Reply:
So , I found an article about this here umich.edu/~eecs381/handouts/IncompleteDeclarations.pdf .
What I did was use forward declaration of struct in platformer_main.h , but I had to put the full definition of it in character_complete_definition.h , which still has the problem of it being a .h file with multiple #includes in it , but it is only used in .cpp files which need to acces the methods of the Character struct and not the other .h files . Is that any better?

Reply:
Actually , I had no idea why I tried getting the Character struct inside of platformer_combat.h ... I didn't need it at all , I  already had the CharacterHealth component that I needed. I fixed the code and returned the includes the way they were before. Sorry about that , I guess it served as a example in todays lecture :D